code/modules/admin/tabs/event_tab.dm

```diff
@@ -148,46 +148,77 @@
-			else return
-		for(var/datum/shuttle/ferry/ert/F in shuttle_controller.process_shuttles)
-			if(F != shuttle)
-				//other ERT shuttles already docked on almayer or about to be
-				if(!F.location || F.moving_status != SHUTTLE_IDLE)
-					if(F.area_station.type == dock_id)
-						message_staff("Warning: That docking zone is already taken by another shuttle. Aborting.")
-						return
-
-		for(var/area/A in all_areas)
-			if(A.type == dock_id)
-				shuttle.area_station = A
-				break
-
-	if(!shuttle.can_launch())
-		message_staff("Warning: Unable to launch this Distress shuttle at this moment. Aborting.")
-		return
-
-	shuttle.launch()
+	var/newShuttle = FALSE
+	if(tag in list("Distress", "Distress_PMC", "Distress_UPP"))
+		newShuttle = TRUE
+	var/list/shuttle_map = list(
+		"Distress"="ert1",
+		"Distress_PMC"="ert2",
+		"Distress_UPP"="ert3",
+	)
+	if(newShuttle)
+		var/shuttleId = shuttle_map[tag]
+		var/list/docks = SSshuttle.stationary
+		var/list/targets = list()
+		var/list/target_names = list()
+		var/obj/docking_port/mobile/emergency_response/ert = SSshuttle.getShuttle(shuttleId)
+		for(var/obj/docking_port/stationary/emergency_response/dock in docks)
+			var/can_dock = ert.canDock(dock)
+			if(can_dock == SHUTTLE_CAN_DOCK)
+				targets += list(dock)
+				target_names +=  list(dock.name)
+		var/dock_name = tgui_input_list(usr, "Where on the [MAIN_SHIP_NAME] should the shuttle dock?", "Select a docking zone:", target_names)
+		var/launched = FALSE
+		for(var/i in targets)
+			var/obj/docking_port/stationary/emergency_response/dock = i
+			if(dock.name == dock_name)
+				var/obj/docking_port/stationary/target = SSshuttle.getDock(dock.id)
+				ert.request(target)
+				launched=TRUE
+		if(!launched)
+			message_staff("Warning: Unable to launch this Distress shuttle at this moment. Aborting.")
+			return
+	else
+		var/datum/shuttle/ferry/ert/shuttle = shuttle_controller.shuttles[tag]
+		if(!shuttle || !istype(shuttle))
+			message_staff("Warning: Distress shuttle not found. Aborting.")
+			return
+
+		if(shuttle.location) //in start zone in admin z level
+			var/dock_id
+			var/dock_list = list("Port", "Starboard", "Aft")
+			if(shuttle.use_umbilical)
+				dock_list = list("Port Hangar", "Starboard Hangar")
+			if(shuttle.use_small_docks)
+				dock_list = list("Port Engineering", "Starboard Engineering")
+			var/dock_name = tgui_input_list(usr, "Where on the [MAIN_SHIP_NAME] should the shuttle dock?", "Select a docking zone:", dock_list)
+			switch(dock_name)
+				if("Port") dock_id = /area/shuttle/distress/arrive_2
+				if("Starboard") dock_id = /area/shuttle/distress/arrive_1
+				if("Aft") dock_id = /area/shuttle/distress/arrive_3
+				if("Port Hangar") dock_id = /area/shuttle/distress/arrive_s_hangar
+				if("Starboard Hangar") dock_id = /area/shuttle/distress/arrive_n_hangar
+				if("Port Engineering") dock_id = /area/shuttle/distress/arrive_s_engi
+				if("Starboard Engineering") dock_id = /area/shuttle/distress/arrive_n_engi
```

Comment:
we generally prefer control statements to not have code on the same line

Reply:
Fair, this was a copy and paste job from elsewhere